front/app/components/admin/TerminologyConfig/index.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+import { mapValues, lowerCase } from 'lodash-es';
+
+// components
+import Collapse from 'components/UI/Collapse';
+import { SectionField } from 'components/admin/Section';
+import InputMultilocWithLocaleSwitcher from 'components/UI/InputMultilocWithLocaleSwitcher';
+import Button from 'components/UI/Button';
+import { ButtonWrapper } from 'components/admin/PageWrapper';
+
+// i18n
+import { injectIntl, MessageDescriptor } from 'utils/cl-intl';
+import { InjectedIntlProps } from 'react-intl';
+import messages from './messages';
+
+// styling
+import styled from 'styled-components';
+
+// typings
+import { Multiloc } from 'typings';
+
+const Container = styled.form`
+  width: 100%;
+  max-width: 500px;
+  padding: 25px;
+  border-radius: ${(props: any) => props.theme.borderRadius};
+  border: solid 1px #ddd;
+  background: #fff;
+`;
+
+interface ISaveParams {
+  singular: Multiloc | undefined;
+  plural: Multiloc | undefined;
+}
+
+interface Props {
+  className?: string;
+  terminologyMessage: MessageDescriptor;
+  tooltipMessage: MessageDescriptor;
+  singularValueMultiloc?: Multiloc;
+  pluralValueMultiloc?: Multiloc;
+  singularLabelMessage: MessageDescriptor;
+  pluralLabelMessage: MessageDescriptor;
+  singularPlaceholderMessage: MessageDescriptor;
+  pluralPlaceholderMessage: MessageDescriptor;
+  onSave: (params: ISaveParams) => void;
+}
+
+type TSubmitState = 'enabled' | 'saving' | 'error' | 'success';
+
+export const getTerm = (
+  localTerm: Multiloc | undefined,
+  configTerm: Multiloc | undefined
+) => localTerm || configTerm || {};
+
+const TerminologyConfig = ({
```

Comment:
Refactored the `AreaTermConfig` into this `TerminologyConfig` component, which is reused in `TopicTermConfig`.